tests/lax_numpy_test.py

```diff
@@ -4889,12 +4889,19 @@ def testArangeJaxpr(self, args, specify_device):
     device = jax.devices()[-1] if specify_device else None
     kwargs = {"device": device}
     jaxpr = jax.make_jaxpr(lambda: jnp.arange(*args, **kwargs))()
-    # We have 2 statements in jaxpr:
-    # [a:i32[5] = iota[dimension=0 dtype=int32 shape=(5,)],
-    #  a:i32[5] = device_put[devices=[None] srcs=[None]] b]
-    num_eqs = 2 if device is not None else 1
-    self.assertEqual(len(jaxpr.jaxpr.eqns), num_eqs)
-    self.assertEqual(jaxpr.jaxpr.eqns[0].primitive, lax.iota_p)
+    self.assertEqual(len(jaxpr.jaxpr.eqns), 1)
```

Comment:
You need to add more tests actually testing the `sharding` bit! Current this test is only testing for SingleDevice stuff.